From bca82e2fba9505f1bffdd5cb4aa23f4bb3e05f84 Mon Sep 17 00:00:00 2001
From: Yushiro FURUKAWA <y.furukawa_2@jp.fujitsu.com>
Date: Tue, 24 Jul 2018 23:39:59 +0900
Subject: [PATCH] Fix broken unittests

_clean() method is renamed to clean() in osprofiler 2.3.0.
It was suggested in a past neutronclient review.

Closes-Bug: #1783789
Co-Authored-By: Akihiro Motoki <amotoki@gmail.com>
Depends-On: https://review.openstack.org/#/c/592018/
Change-Id: Ic8e03db85dc08cfdcac5507e99ecdf7eac8eb972
---
 lower-constraints.txt                 | 2 +-
 neutronclient/tests/unit/test_http.py | 2 +-
 test-requirements.txt                 | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lower-constraints.txt b/lower-constraints.txt
index 81394e2..2bf9a31 100644
--- a/lower-constraints.txt
+++ b/lower-constraints.txt
@@ -63,7 +63,7 @@ oslo.serialization==2.18.0
 oslo.service==1.24.0
 oslo.utils==3.33.0
 oslotest==3.2.0
-osprofiler==1.4.0
+osprofiler==2.3.0
 paramiko==2.0.0
 Paste==2.0.2
 PasteDeploy==1.5.0
diff --git a/neutronclient/tests/unit/test_http.py b/neutronclient/tests/unit/test_http.py
index 21b6155..d76e9bc 100644
--- a/neutronclient/tests/unit/test_http.py
+++ b/neutronclient/tests/unit/test_http.py
@@ -77,7 +77,7 @@ class TestHTTPClientMixin(object):
 
     def test_osprofiler_headers_are_injected(self):
         osprofiler.profiler.init('SWORDFISH')
-        self.addCleanup(osprofiler.profiler._clean)
+        self.addCleanup(osprofiler.profiler.clean)
 
         headers = {'Accept': 'application/json'}
         headers.update(osprofiler.web.get_trace_id_headers())
diff --git a/test-requirements.txt b/test-requirements.txt
index eb3bcf1..77a33e6 100644
--- a/test-requirements.txt
+++ b/test-requirements.txt
@@ -9,7 +9,7 @@ fixtures>=3.0.0 # Apache-2.0/BSD
 flake8-import-order==0.12 # LGPLv3
 mock>=2.0.0 # BSD
 oslotest>=3.2.0 # Apache-2.0
-osprofiler>=1.4.0 # Apache-2.0
+osprofiler>=2.3.0 # Apache-2.0
 python-openstackclient>=3.12.0 # Apache-2.0
 python-subunit>=1.0.0 # Apache-2.0/BSD
 requests-mock>=1.2.0 # Apache-2.0
-- 
2.18.0

